### ISSUE TITLE
jaxb parser does not handle JsonSubTypes as well as it used to

### ISSUE THREAD

**Issue by thesam** [vojtechhabarta/typescript-generator#794]
Hi!

My team is using typescript-generator v2.23.603. We want to use the latest version but can't since the behaviour for JsonSubTypes has changed for the jaxb parser. (We use the jaxb parser since we use also use JAXB XmlElement annotations in our Java code). Is there some configuration we can change to get the old behaviour?

The following code snippet can be used to reproduce the problem (I added it to TaggedUnionsTest in my working copy):
```
    @Test
    public void reproduce() {
        final Settings settings = TestUtils.settings();
        settings.jsonLibrary = JsonLibrary.jaxb;
        settings.quotes = "'";
        final String output = new TypeScriptGenerator(settings).generateTypeScript(Input.from(Parent.class));
        System.out.println(output);
    }

    @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
    @JsonSubTypes({@JsonSubTypes.Type(value=Child1.class,name="one"), @JsonSubTypes.Type(value = Child2.class,name = "two")})
    private static class Parent {

    }

    private static class Child1 extends Parent {

    }

    private static class Child2 extends Parent{}
```
Output from v2.23.603. The "type" discriminator is generated with the type **"one" | "two"**, this is great.
```
interface Parent {
    type: "one" | "two";
}

interface Child1 extends Parent {
    type: "one";
}

interface Child2 extends Parent {
    type: "two";
}

type ParentUnion = Child1 | Child2;
```
Output from v2.35.1026. The "type" discriminator is generated with the type **string** instead.
```
interface Parent {
    type: string;
}

interface Child1 extends Parent {
    type: string;
}

interface Child2 extends Parent {
    type: string;
}

type ParentUnion = Child1 | Child2;
```

**Comment by vojtechhabarta** [comment]
Fix released in [3.2.1263](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v3.2.1263).

**Comment by dan-lugg** [comment]
@vojtechhabarta Is this behavior configurable? I'm using the Gradle plugin, and would prefer (for a particular use-case) to keep the type discriminator field "open", as in just emit `string`, rather than a union of all type names.
